src/ar/tuktukcinema/src/eu/kanade/tachiyomi/animeextension/ar/tuktukcinema/Tuktukcinema.kt

```diff
@@ -151,6 +159,23 @@ class Tuktukcinema : ConfigurableAnimeSource, ParsedAnimeHttpSource() {
 
     private fun extractVideos(url: String): List<Video> {
         return when {
+            url.contains("egtpgrvh") -> {
+                val videoList = mutableListOf<Video>()
+                val request = client.newCall(GET(url, headers)).execute().asJsoup()
+                val data = request.selectFirst("script:containsData(m3u8)")!!.data()
+                val masterUrl = data.substringAfter("sources: [{").substringAfter("file:\"").substringBefore("\"}")
+                val m3u8PrimeLink = masterUrl.substringBefore("master.m3u8")
+                val masterPlaylist = client.newCall(
+                    GET(masterUrl, headers),
+                ).execute().body.string()
+                val separator = "#EXT-X-STREAM-INF:"
+                masterPlaylist.substringAfter(separator).split(separator).forEach {
+                    val quality = it.substringAfter("RESOLUTION=").substringAfter("x").substringBefore(",") + "p "
+                    val videoUrl = m3u8PrimeLink + it.substringAfter("\n").substringBefore("\n")
+                    videoList.add(Video(videoUrl, quality, videoUrl, headers))
+                }
+                return videoList
```

Comment:
Doing it like this is not really a bad thing, but to avoid duplicated code you add a dependency on the `implementation(project(':lib-playlist-utils'))` lib and then it would be:
```suggestion
                PlaylistUtils(client, headers).extractFromHls(masterUrl)
```

Reply:
is this new lib for m3u8 to extract qualities ohh